examples/taskruns/step-script.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: tekton.dev/v1alpha1
+kind: TaskRun
+metadata:
+  name: step-script
+spec:
+  taskSpec:
+    steps:
+    - image: ubuntu
+      env:
+      - name: FOO
+        value: foooooooo
+      script:
+      - '#!/bin/bash'
```

Comment:
This makes me wonder whether `Script` shouldn't just be a `string`, which can contain newlines.

That would change these to:

```yaml
steps:
- image: ubuntu
  env:
  - name: FOO
    value: foooooooo
  script: |
    #!/bin/bash
    set -euxo pipefail
    echo "Hello from Bash!"
    echo FOO is ${FOO}
    for i in {1..10}; do
      echo line $i
    done
- image: node
  script: |
    #!/usr/bin/env node
    console.log("Hello from Node!")
- image: python
  script: |
    #!/usr/bin/env python3
    print("Hello from Python!")
- image: perl
  script: |
    #!/usr/bin/perl
    print "Hello from Perl!"
```

...which I think is a bit clearer. We'd only lose the ability to have YAML comments inline, but that's probably not worth it.